tests/jax/models/MNIST_CNN/test_mnistcnn.py

```diff
@@ -0,0 +1,89 @@
+# SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Dict, Sequence
+
+import jax
+import jax.numpy as jnp
+import pytest
+import functools
+from flax import linen as nn
+from infra import ComparisonConfig, ModelTester, RunMode
+from transformers import AutoTokenizer, FlaxDistilBertForMaskedLM
+
+from CNN import CNN
+
+class MNISTCNNTester(ModelTester):
+    """Tester for MNIST CNN model."""
+
+    # @override
+    @staticmethod
+    def _get_model() -> nn.Module:
+        return CNN()
+
+    # @override
+    @staticmethod
+    def _get_forward_method_name() -> str:
+        return "apply"
+
+    # @override
+    @staticmethod
+    def _get_input_activations() -> Sequence[jax.Array]:
+        img = jnp.ones((28, 28, 1))
+        return img
+    
+    # @override
+    def _get_forward_method_args(self):
+        inp = self._get_input_activations()
```

Comment:
I left it like that to make it super clear that that method should only be concerned with forming input activations, maybe by fetching them from somewhere, or making raw tensors, images, whatever.

We cannot know (although it most certainly will be the first one) which argument to model's forward method is input activations, nor its name. For example in `FlaxDistilBertPreTrainedModel`  input activations are called `input_ids` which is not super intuitive for me, although it is the first arg in `__call__`.  

Thus I left this to be as flexible as possible. 

We could maybe make default behaviour of `_get_forward_method_args` be

```python
def _get_forward_method_args(self) -> Sequence[Any]:
    return [self._get_input_activations()]    # instead of []
```

if that makes any sense. 

But you can't do something like this for `_get_forward_method_kwargs` since you never know the name of the arg refering to input activations, as I showed above. Thus I'd leave it as is since it enforces you to be explicit and carefully respect model's call interface.

Reply:
A convention(abeit an often broken one) in JAX broadly is to have model take parameters first. Having inputs as the first argument is not a good default. It also has the benefit of not getting in the way if the user wants to be explicit and only go trough kwargs.

Reply:
Got it, makes sense.